querybook/webapp/components/QueryEditor/QueryEditor.tsx

```diff
@@ -408,6 +450,42 @@ export const QueryEditor: React.FC<
             [isTokenInTable, matchFunctionWithDefinition, openTableModal]
         );
 
+        const onTextHoverShortDebounce = useMemo(
+            () =>
+                debounce((editor: CodeMirror.Editor, node, e, pos, _token) => {
+                    if (markerRef.current == null) {
+                        const suggestionAnnotation =
+                            getSuggestionByPosition(pos);
+                        if (suggestionAnnotation != null) {
+                            markTextAndShowTooltip(
+                                editor,
+                                suggestionAnnotation.from,
+                                suggestionAnnotation.to,
+                                {
+                                    onAcceptSuggestion: () =>
+                                        editor.replaceRange(
+                                            suggestionAnnotation.suggestion,
+                                            suggestionAnnotation.from,
+                                            suggestionAnnotation.to
+                                        ),
+                                    suggestionText:
+                                        suggestionAnnotation.suggestion,
+                                }
+                            );
+                        }
+                    }
+                }, 100),
+            [getSuggestionByPosition]
+        );
+
+        const onTextHover = useCallback(
+            async (editor: CodeMirror.Editor, node, e, pos, token) => {
+                onTextHoverLongDebounce(editor, node, e, pos, token);
+                onTextHoverShortDebounce(editor, node, e, pos, token);
```

Comment:
why we need to call two debounce functions here? can you add some comments to explain it?

Reply:
added comments!